lib/src/components/stack/Ancestor_Stack.res

```diff
@@ -1 +1,715 @@
+module Make = (Maker: Ancestor_StylesMaker.T) => {
+  module Styles = Ancestor_Styles.Make(Maker)
+  module Base = Ancestor_Base.Make(Maker)
 
+  type direction = [
+    | #vertical
+    | #horizontal
+  ]
+
+  /**
+   * Specific styles for the API.
+   */
+  let createStack = (~direction=?, ~spacing=?, ()) => {
+    let baseStyles = `
+      display: flex;
+    `
+
+    let responsiveStyles = [
+      Styles.createResponsiveProp(~prop=spacing, spacing =>
+        `gap: ${Styles.Spacing.make(spacing)};`
+      ),
+      Styles.createResponsiveProp(
+        ~prop=direction,
+        ~defaultStyles=`flex-direction: row;`,
+        direction =>
+          switch direction {
+          | #vertical => "flex-direction: row;"
+          | #horizontal => "flex-direction: column"
+          },
+      ),
+    ]
+
+    Maker.css(
+      `
+      ${baseStyles}
+      ${responsiveStyles->Styles.merge}
+      `,
+    )
+  }
+
+  let renderWithDivider = (children, divider) => {
+    let total = React.Children.count(children)
+    children->React.Children.mapWithIndex((element, index) => {
+      if index == total - 1 {
+        element
+      } else {
+        <> element divider </>
+      }
+    })
+  }
+
+  @react.component
+  let make = (
+    // Stack props
+    ~direction: option<Styles.responsiveProp<direction>>=?,
```

Comment:
Hmm, I'm not sure too. I thought to use Stack with a direction option to make the stack horizontal or a vertical instead of creating two components. Reading the Braid's documentation I haven't found much difference between Inline and Columns. What do you think? Is it worth the effort creating two separated components? My reference was Chakra's [Stack](https://chakra-ui.com/docs/components/stack/usage) component.

Reply:
But I like the idea of creating very specialized components/primitives like Inline. Btw, I was thinking about to create primivites like [Center](https://chakra-ui.com/docs/components/center) or [Wrap](https://chakra-ui.com/docs/components/wrap).

Reply:
In Braid, Inline is used for laying out an arbitrary number of arbitrary sized elements.

A horizontal Stack only makes sense in my mind if the parent container is horizontally scrollable otherwise you're basically adding a different API to the `Inline` or `Columns` primitives. At this point it feels like a different component entirely and so shouldn't be called Stack IMO.

In other words I think a core rule of the Stack primitive is that you can list an arbitrary number of elements and they'll all lay next to each other in a single direction - this isn't the case right now with this horizontal implementation I don't think?

Reply:
> A horizontal Stack only makes sense in my mind if the parent container is horizontally scrollable otherwise you're basically adding a different API to the Inline or Columns primitives. At this point it feels like a different component entirely and so shouldn't be called Stack IMO.

But sometimes you want to stack elements horizontally, right? I'm not sure but about the scroll behavior I think is something that you can deal when you're creating the interface, sometimes you need scroll and sometimes you don't.
Columns (Grid in Ancestor) is more focused on creating components/interfaces based on 12-columns, I mean, when you need to set a size (in columns) for elements, etc.

Stack is about (at least to me) to stack elements, sometimes its sized (explicit setting width for example) and sometimes its not. It's just a primitive to stack elements (vertically or horizontally IMO).

What do you think? This implementation works for you? Since you're mentioned in #2 that Stack is a must 😄